### PR TITLE
Enable CUDA support in cmake when available

### PR DESCRIPTION
**Pull request by rwy0717**
Signed-off-by: Robert Young <rwy0717@gmail.com>

### CODE REVIEW

File: cmake/config.cmake [hunk 1]
```diff
@@ -193,7 +193,6 @@ set(OMR_NOTIFY_POLICY_CONTROL OFF CACHE BOOL "TODO: Document")
 
 set(OMR_ENV_GCC OFF CACHE BOOL "TODO: Document")
 
-
-set(OMR_OPT_CUDA OFF CACHE BOOL "TODO: Document")
+set(OMR_OPT_CUDA ${CUDA_FOUND} CACHE BOOL "Enable CUDA support in OMR")
```
Comment (charliegracie) [inline cmake/config.cmake]: If some forces CUDA off via a cache or command line option will this override it?

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by charliegracie** [comment]
@genie-omr build x32linux

**Comment by charliegracie** [comment]
@rwy0717 the 32 bit CMake build failed 2 times with this change.  It is failing in CMake to find pthreads.  It failed on 2 different machines.  I do not see how your change is related but all other builds passed on 32 bit linux today.

File: cmake/config.cmake (hunk 1 above)
Comment (rwy0717) [reply cmake/config.cmake]: Yep

**Comment by rwy0717** [comment]
It's my fault: The CUDA package requires that the Threads package is found, even if CUDA is optional. I think I can work around this pretty easily.

The CUDA package was deprecated in CMake version 3.10. If we can upgrade our minimum cmake version to 3.9, there is a different, better supported way to consume CUDA. Usage would look like this:

```cmake
include(CheckLanguage)

check_language(CUDA)

if (CMAKE_CUDA_COMPILER)
  enable_language(CUDA)
  set(OMR_OPT_CUDA ON)
else()
  message("OMR: No CUDA support")
endif()

if (OMR_OPT_CUDA)
  target_include_directories(omrport_obj
    ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
  )
endif()
```

Is there any chance of getting 3.9, or should I stick to the older way of getting CUDA?

**Comment by rwy0717** [comment]
Updated the PR, so If Threads is not found, don't try to find CUDA.

File: cmake/config.cmake (hunk 1 above)
Comment (charliegracie) [reply cmake/config.cmake]: hmmm I think it is incorrect for an automatic detection to override a cache/command line option

**Comment by charliegracie** [comment]
@dnakamura what is the version of CMake that we are likely going to use on zOS?

File: cmake/config.cmake (hunk 1 above)
Comment (rwy0717) [reply cmake/config.cmake]: Oh sorry, I misinterpreted your sentence. The user-provided value would be used, overriding the automatically determined default value.

**Comment by rwy0717** [comment]
I checked with him, he believes it's 3.4, so I'm sticking with the original implementation, with the additional workarounds for finding Thread.

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by rwy0717** [comment]
Build failure is because we're detecting cuda on x86, but OMR doesn't support cuda on 32bit linux. I'm going to add an additional test for 64 bit before trying to enable CUDA.

File: cmake/config.cmake (hunk 1 above)
Comment (charliegracie) [reply cmake/config.cmake]: ok sounds much better.

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by youngar** [comment]
LGTM. It looks like enabling the OMR_OPT_CUDA flag is all we need.

**Pull request merged**